ambari-server/src/main/resources/common-services/HDFS/2.1.0.2.0/package/alerts/alert_metrics_deviation.py

```diff
@@ -233,7 +234,7 @@ def execute(configurations={}, parameters={}, host_name=None):
 
     kinit_timer_ms = parameters.get(KERBEROS_KINIT_TIMER_PARAMETER, DEFAULT_KERBEROS_KINIT_TIMER_MS)
 
-    name_service = configurations[NAMESERVICE_KEY]
+    name_service = get_name_service_by_hostname(hdfs_site, host_name)
```

Comment:
@majorendre Won't this fail in a non HA, non Federated setup? 

Based on the code
`if not name_services_string:
    raise Fail('Not a HA setup')`

Reply:
By contract this is an HA utility. I think it is better to raise an error then returning a None. In the alert_metrics_deviation.py it is called on a HA only branch. There are unit tests covering both HA and non HA cases.

Reply:
Raising error scenario needs to be tested to not break something in agent code if it is not expected to be throwing an error. With contract-less languages like python these things are hard to do, although I see your point and it's valid, I believe that is what @avijayanhwx is concerned about.

Reply:
It is only called from the alert_metrics_deviation.py. That one is tested, see test_alert_metrics_deviation.py.